plugins/treeview/prism-treeview.js

```diff
@@ -0,0 +1,60 @@
+Prism.languages.treeview = {
+	'treeview-part': {
+		pattern: /^.+/m,
+		inside: {
+			'entry-line': [
+				{
+					pattern: /\|-- |├── /,
+					alias: 'line-h'
+				},
+				{
+					pattern: /\|   |│   /,
+					alias: 'line-v'
+				},
+				{
+					pattern: /`-- |└── /,
+					alias: 'line-v-last'
+				},
+				{
+					pattern: / {4}/,
+					alias: 'line-v-gap'
+				}
+			],
+			'entry-name': {
+				pattern: /.*\S.*/,
+				inside: {
+					// symlink
+					'operator': / -> /,
+				}
+			}
+		}
+	}
+};
+
+Prism.hooks.add('wrap', function (env) {
+	if (env.language === 'treeview') {
+		if (env.type === 'entry-name') {
+			if (/(^|[^\\])\/\s*$/.test(env.content)) {
+				env.content = env.content.slice(0, -1);
+				// This is a folder
+				env.classes.push('dir');
+			} else {
+
+				if (/(^|[^\\])[=*|]\s*$/.test(env.content)) {
+					env.content = env.content.slice(0, -1);
+				}
+
+				var parts = env.content.toLowerCase().replace(/\s+/, '').split('.');
+				while (parts.length > 1) {
+					parts.shift();
+					// Ex. 'foo.min.js' would become '<span class="token keyword ext-min-js ext-js">foo.min.js</span>'
+					env.classes.push('ext-' + parts.join('-'));
```

Comment:
maybe add also a `ext-js` here with only the last extension if someone use points in normal filename like: `this.is.my.file.html.js`

Reply:
@muescha This is already the case I believe. In the example you gave, the token would have the classes
```
ext-is-my-file-html-js ext-my-file-html-js ext-html-js ext-js
```

Reply:
It behaves as described @Golmote. I think it's ok to add few more classes than necessary in these edge cases.

Reply:
ah yes 🤦‍♂ its a loop :) 

great :)